gossip: negative WaitGroup counter

Seen in [this build](https://circleci.com/gh/petermattis/cockroach/698?utm_campaign=build-failed&utm_medium=email&utm_source=notification).

```
panic: sync: negative WaitGroup counter

goroutine 5284 [running]:
panic(0x14cb6e0, 0xc820cb56f0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
sync.(*WaitGroup).Add(0xc820ce3b10, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:71 +0xd0
sync.(*WaitGroup).Done(0xc820ce3b10)
	/usr/local/go/src/sync/waitgroup.go:96 +0x2a
github.com/cockroachdb/cockroach/storage_test.TestStoreRangeUpReplicate.func1(0xc820cb5510, 0x7, 0xc820bfa210, 0x27, 0x30, 0x1447b6277f34e803, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:544 +0x20
github.com/cockroachdb/cockroach/gossip.(*infoStore).runCallbacks.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:262 +0xaa
github.com/cockroachdb/cockroach/gossip.(*infoStore).runCallbacks.func2()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:284 +0x129
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc82022c0e0, 0x209b0e7, 0x13, 0x11e, 0xc820cb5530)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:181 +0x62
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:182 +0xc9

```

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -30,7 +30,6 @@ import (
 
 	"github.com/cockroachdb/cockroach/client"
 	"github.com/cockroachdb/cockroach/config"
-	"github.com/cockroachdb/cockroach/gossip"
 	"github.com/cockroachdb/cockroach/keys"
 	"github.com/cockroachdb/cockroach/roachpb"
 	"github.com/cockroachdb/cockroach/storage"
@@ -538,14 +537,17 @@ func TestStoreRangeUpReplicate(t *testing.T) {
 	defer mtc.Stop()
 
 	// Initialize the gossip network.
-	var wg sync.WaitGroup
-	wg.Add(len(mtc.stores))
-	key := gossip.MakePrefixPattern(gossip.KeyStorePrefix)
-	mtc.stores[0].Gossip().RegisterCallback(key, func(_ string, _ roachpb.Value) { wg.Done() })
+	storeDescs := make([]*roachpb.StoreDescriptor, 0, len(mtc.stores))
 	for _, s := range mtc.stores {
-		s.GossipStore()
+		desc, err := s.Descriptor()
+		if err != nil {
+			t.Fatal(err)
+		}
+		storeDescs = append(storeDescs, desc)
 	}
-	wg.Wait()
+	for _, g := range mtc.gossips {
+		gossiputil.NewStoreGossiper(g).GossipStores(storeDescs, t)
+	}
 
 	// Once we know our peers, trigger a scan.
 	mtc.stores[0].ForceReplicationScanAndProcess()
@@ -642,14 +644,17 @@ func TestStoreRangeDownReplicate(t *testing.T) {
 	mtc.replicateRange(desc.RangeID, 3, 4)
 
 	// Initialize the gossip network.
-	var wg sync.WaitGroup
-	wg.Add(len(mtc.stores))
-	key := gossip.MakePrefixPattern(gossip.KeyStorePrefix)
-	mtc.stores[0].Gossip().RegisterCallback(key, func(_ string, _ roachpb.Value) { wg.Done() })
+	storeDescs := make([]*roachpb.StoreDescriptor, 0, len(mtc.stores))
 	for _, s := range mtc.stores {
-		s.GossipStore()
+		desc, err := s.Descriptor()
+		if err != nil {
+			t.Fatal(err)
+		}
+		storeDescs = append(storeDescs, desc)
 	}
-	wg.Wait()
+	for _, g := range mtc.gossips {
+		gossiputil.NewStoreGossiper(g).GossipStores(storeDescs, t)
+	}
 
 	maxTimeout := time.After(10 * time.Second)
 	succeeded := false
@@ -1234,11 +1239,8 @@ func TestStoreRangeRebalance(t *testing.T) {
 		}
 		storeDescs = append(storeDescs, desc)
 	}
-	var sgs []*gossiputil.StoreGossiper
 	for _, g := range mtc.gossips {
-		sg := gossiputil.NewStoreGossiper(g)
-		sgs = append(sgs, sg)
-		sg.GossipStores(storeDescs, t)
+		gossiputil.NewStoreGossiper(g).GossipStores(storeDescs, t)
 	}
 
 	// This can't use SucceedsSoon as using the exponential backoff mechanic
@@ -1329,9 +1331,9 @@ func TestReplicateRogueRemovedNode(t *testing.T) {
 
 	// Try to issue a command on node 2. It should not be able to commit
 	// (so we add it asynchronously).
-	startWG := sync.WaitGroup{}
+	var startWG sync.WaitGroup
 	startWG.Add(1)
-	finishWG := sync.WaitGroup{}
+	var finishWG sync.WaitGroup
 	finishWG.Add(1)
 	go func() {
 		rng, err := mtc.stores[2].GetReplica(raftID)

